plugins/modules/idrac_network_attributes.py

```diff
@@ -0,0 +1,534 @@
+        if rf_settings:
+            if 'Maintenance' in aplytm:
+                if aplytm not in rf_settings:
+                    self.module.exit_json(failed=True, msg=APPLY_TIME_NOT_SUPPORTED_MSG.format(aplytm))
+                else:
+                    rf_set['ApplyTime'] = aplytm
+                    m_win = self.module.params.get('maintenance_window')
+                    self.__validate_time(m_win.get('start_time'))
+                    rf_set['MaintenanceWindowStartTime'] = m_win.get('start_time')
+                    rf_set['MaintenanceWindowDurationInSeconds'] = m_win.get('duration')
+            else:  # assuming OnReset is always
+                if aplytm == "Immediate" and aplytm not in rf_settings:
+                    reboot_req = True
+                    aplytm = 'OnReset'
+                rf_set['ApplyTime'] = aplytm
+        return rf_set, reboot_req
+
+    def get_current_server_registry(self):
+        reg = {}
+        oem_network_attributes = self.module.params.get('oem_network_attributes')
+        firm_ver = self.__get_idrac_firmware_version()
+        if oem_network_attributes:
+            if LooseVersion(firm_ver) >= '6.0':
+                oem_links = get_dynamic_uri(self.idrac, self.network_device_function_id_uri, 'Links')
+                uri = oem_links.get('Oem').get('Dell').get('DellNetworkAttributes').get('@odata.id')
+                reg = get_dynamic_uri(self.idrac, uri).get('Attributes', {})
+            if '3.0' < LooseVersion(firm_ver) < '6.0':
+                reg = self.__get_registry_fw_less_than_6_more_than_3()
+        else:  # For Redfish
+            pass
+        return reg
+
+    def extract_error_msg(self, resp):
+        error_info = {}
+        error = resp.json_data.get('error')
+        for each_dict_err in error.get("@Message.ExtendedInfo"):
+            key = each_dict_err.get('MessageArgs')[0]
+            msg = each_dict_err.get('Message')
+            if key not in error_info:
+                error_info.update({key: msg})
+        return error_info
+
+    def get_diff_between_current_and_module_input(self, module_attr, server_attr):
+        invalid = {}
+        diff = recursive_diff(module_attr, server_attr)
+        for each_attr in module_attr:
+            if each_attr not in server_attr:
+                invalid.update({each_attr: ATTRIBUTE_NOT_EXIST_CHECK_IDEMPOTENCY_MODE})
+        return diff, invalid
+
+    def perform_validation_for_network_adapter_id(self):
+        odata = '@odata.id'
+        network_adapter_id = self.module.params.get('network_adapter_id')
+        found_adapter = False
+        first_resource_id_uri = self.__get_resource_id()
+        network_adapters = get_dynamic_uri(self.idrac, first_resource_id_uri, 'NetworkAdapters')[odata]
+        network_adapter_list = get_dynamic_uri(self.idrac, network_adapters, 'Members')
+        for each_adapter in network_adapter_list:
+            if network_adapter_id in each_adapter.get(odata, ''):
```

Comment:
Any specific reason why are we passing empty string here - each_adapter.get(odata, '').By default its None which is also behave the same way.

Reply:
Whole code is written dynamically, here each_adapter is each_adapter's uri which is string, that's why it's has mentioned, nothing to do with what user is passing.